managed/services/management/dbaas/kubernetes_server.go

```diff
@@ -490,25 +517,68 @@ func (k kubernetesServer) installDefaultOperators(operatorsToInstall map[string]
 		if !ok || channel == "" {
 			channel = "stable-v0"
 		}
-		params := kubernetes.InstallOperatorRequest{
-			Namespace:              namespace,
-			Name:                   operatorName,
-			OperatorGroup:          operatorGroup,
-			CatalogSource:          "percona-dbaas-catalog",
-			CatalogSourceNamespace: catalogSourceNamespace,
-			Channel:                channel,
-			InstallPlanApproval:    v1alpha1.ApprovalManual,
-		}
+		_, err := kubeClient.GetSubscription(ctx, namespace, operatorName)
+		if err == nil || apierrors.IsNotFound(err) {
+			params := kubernetes.InstallOperatorRequest{
+				Namespace:              namespace,
+				Name:                   operatorName,
+				OperatorGroup:          operatorGroup,
+				CatalogSource:          "percona-dbaas-catalog",
+				CatalogSourceNamespace: catalogSourceNamespace,
+				Channel:                channel,
+				InstallPlanApproval:    v1alpha1.ApprovalManual,
+			}
 
-		if err := kubeClient.InstallOperator(ctx, params); err != nil {
-			retval["vm"] = err
-			k.l.Errorf("cannot instal PXC operator in the new cluster: %s", err)
+			if err := kubeClient.InstallOperator(ctx, params); err != nil {
+				retval["vm"] = err
+				k.l.Errorf("cannot instal PXC operator in the new cluster: %s", err)
+			}
 		}
 	}
 
 	return retval
 }
 
+// uninstallVM unistalls Victoria Metrics Operator.
+// To delete the vm operator we must:
+//  1. Delete the subscription.
+//  2. Delete the CSV.
+func uninstallVM(ctx context.Context, kubeClient kubernetesClient, subscription *v1alpha1.Subscription) error {
+	key := types.NamespacedName{
```

Comment:
Actually, I'm not sure that's a great option because there can be already existing VM agents running in the cluster (e.g. we're registering k8s cluster with the 4 db clusters and one VM agent running. In that case we may have broken monitoring for these database clusters). 

Yet, I don't have any recommendations here. Maybe getting running vmagents and removing agents only and reconfiguring them would be a better solution. What do you think?

Reply:
How can we know if the agent that's already installed belongs to the same PMM server? 
Does it make sense? Maybe we shouldn't do anything and it should be only removed when you unregister the K8s server. 
I also added a new label `app.kubernetes.io/part-of:pmm` to be able to check that the operator/s were installed by pmm. Maybe we could also use `app.kubernetes.io/managed-by` with the PMM server id and only uninstall things that were installed by this PMM installation. 

DOC REF: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/